### PR TITLE
Port to blender4 Merged with blender4-image-export

### PR DESCRIPTION
**Pull request by Spiderguy-F**
Opened a new pull request, like you suggested

### CODE REVIEW

**Comment by keianhzo** [comment]
I think this includes many commits that are already part of the main branch. Maybe you have merged the other fixes branch on this one? I'd expect this PR to only have the additional changed. Let's talk about it int he meetup.

**Review by keianhzo — COMMENTED** [review]
As talked in the meeting, we need to correctly handle the return parameters from `__gather_texture_transform_and_tex_coord` and we should be good to land.

I think probably something like this would work:
```
if bpy.app.version < (3, 2, 0):
    tex_transform, tex_coord = gltf2_blender_gather_texture_info.__gather_texture_transform_and_tex_coord(
        texture_socket, export_settings)
elif bpy.app.version >= (4, 0, 0):
    socket = gltf2_blender_search_node_tree.NodeSocket(
        texture_socket, blender_material)
    tex_transform, tex_coord = gltf2_blender_gather_texture_info.__gather_texture_transform_and_tex_coord(
        socket, export_settings)
else:
    tex_transform, tex_coord, _ = gltf2_blender_gather_texture_info.__gather_texture_transform_and_tex_coord(
        texture_socket, export_settings)
```

File: addons/io_hubs_addon/io/utils.py [hunk 1]
```diff
@@ -366,8 +371,13 @@ def gather_lightmap_texture_info(blender_material, export_settings):
     if bpy.app.version < (3, 2, 0):
         tex_transform, tex_coord = gltf2_blender_gather_texture_info.__gather_texture_transform_and_tex_coord(
             texture_socket, export_settings)
+    elif bpy.app.version >= (4, 0, 0):
+        socket = gltf2_blender_search_node_tree.NodeSocket(texture_socket, blender_material)
+        tex_transform, tex_coord = gltf2_blender_gather_texture_info.__gather_texture_transform_and_tex_coord(
+            socket, export_settings)
     else:
-        tex_transform, tex_coord, _ = gltf2_blender_gather_texture_info.__gather_texture_transform_and_tex_coord(
+        print("processing __gather_texture_transform_and_tex_coord")
```
Comment (keianhzo) [inline addons/io_hubs_addon/io/utils.py:379]:
```suggestion
```

File: addons/io_hubs_addon/io/utils.py [hunk 2]
```diff
@@ -366,8 +371,13 @@ def gather_lightmap_texture_info(blender_material, export_settings):
...
+        tex_transform, tex_coord = gltf2_blender_gather_texture_info.__gather_texture_transform_and_tex_coord(
```
Comment (keianhzo) [inline addons/io_hubs_addon/io/utils.py:380]:
```suggestion
        tex_transform, tex_coord, _ = gltf2_blender_gather_texture_info.__gather_texture_transform_and_tex_coord(
        texture_socket, export_settings)
```

**Review by keianhzo — CHANGES_REQUESTED** [review]
Added updates as commit suggestion

File: addons/io_hubs_addon/io/utils.py [hunk 3]
```diff
@@ -47,7 +49,10 @@ def from_blender_image(image: bpy.types.Image):
 
     def encode(self, mime_type: Optional[str], export_settings) -> Union[Tuple[bytes, bool], bytes]:
         if mime_type == "image/vnd.radiance":
-            return self.encode_from_image_hdr(self.blender_image())
+            if bpy.app.version < (4, 0, 0):
```
Comment (Exairnous) [inline addons/io_hubs_addon/io/utils.py:52]:
```suggestion
            if bpy.app.version < (4, 1, 0):
```

File: addons/io_hubs_addon/io/utils.py [hunk 4]
```diff
@@ -366,8 +371,13 @@ def gather_lightmap_texture_info(blender_material, export_settings):
     if bpy.app.version < (3, 2, 0):
         tex_transform, tex_coord = gltf2_blender_gather_texture_info.__gather_texture_transform_and_tex_coord(
             texture_socket, export_settings)
+    elif bpy.app.version >= (4, 0, 0):
+        socket = gltf2_blender_search_node_tree.NodeSocket(texture_socket, blender_material)
+        tex_transform, tex_coord = gltf2_blender_gather_texture_info.__gather_texture_transform_and_tex_coord(
+            socket, export_settings)
```
Comment (Exairnous) [inline addons/io_hubs_addon/io/utils.py:377]:
```suggestion
    elif bpy.app.version < (4, 0, 0):
        tex_transform, tex_coord, _ = gltf2_blender_gather_texture_info.__gather_texture_transform_and_tex_coord(
            texture_socket, export_settings)
    elif bpy.app.version < (4, 1, 0):
        tex_transform, tex_coord = gltf2_blender_gather_texture_info.__gather_texture_transform_and_tex_coord(
            texture_socket, export_settings)
```

File: addons/io_hubs_addon/io/utils.py [hunk 5]
```diff
@@ -366,8 +371,13 @@ def gather_lightmap_texture_info(blender_material, export_settings):
...
     else:
-        tex_transform, tex_coord, _ = gltf2_blender_gather_texture_info.__gather_texture_transform_and_tex_coord(
+        print("processing __gather_texture_transform_and_tex_coord")
+        tex_transform, tex_coord = gltf2_blender_gather_texture_info.__gather_texture_transform_and_tex_coord(
             texture_socket, export_settings)
```
Comment (Exairnous) [inline addons/io_hubs_addon/io/utils.py:381]:
```suggestion
        socket = gltf2_blender_search_node_tree.NodeSocket(texture_socket, blender_material)
        tex_transform, tex_coord = gltf2_blender_gather_texture_info.__gather_texture_transform_and_tex_coord(
            socket, export_settings)

```

**Review by Exairnous — CHANGES_REQUESTED** [review]
@keianhzo 4.0 support turned out to be trickier than we thought at the dev meetup.

@Spiderguy-F We identified at the meetup that Blender 4.0 was the cause of the errors I had been seeing.  I have added suggestions to fix all the errors and so now the add-on should work for all the supported Blender versions.  I have incorporated the fixes suggested by @keianhzo in my suggestions.  It turns out that Blender 4.0 requires the same code path as versions below 3.2, but I don't see a cleaner way to implement this than what I've suggested here; however, anyone is welcome to try to improve on my suggestions. :)

File: addons/io_hubs_addon/io/utils.py [hunk 6]
```diff
@@ -10,6 +10,8 @@
     from io_scene_gltf2.blender.exp import gltf2_blender_gather_texture_info, gltf2_blender_export_keys
     from io_scene_gltf2.blender.exp import gltf2_blender_image
 from io_scene_gltf2.blender.exp.gltf2_blender_gather_cache import cached
+if bpy.app.version >= (4, 0, 0):
```
Comment (Exairnous) [inline addons/io_hubs_addon/io/utils.py:13]:
```suggestion
if bpy.app.version >= (4, 1, 0):
```

File: addons/io_hubs_addon/io/utils.py [hunk 7]
```diff
@@ -363,12 +368,11 @@ def gather_lightmap_texture_info(blender_material, export_settings):
     # TODO this assumes a single image directly connected to the socket
     blender_image = texture_socket.links[0].from_node.image
     texture = gather_texture(blender_image, export_settings)
-    if bpy.app.version < (3, 2, 0):
-        tex_transform, tex_coord = gltf2_blender_gather_texture_info.__gather_texture_transform_and_tex_coord(
-            texture_socket, export_settings)
-    else:
-        tex_transform, tex_coord, _ = gltf2_blender_gather_texture_info.__gather_texture_transform_and_tex_coord(
-            texture_socket, export_settings)
+    socket = lightmap_node.inputs.get("Lightmap") if bpy.app.version < (4, 0, 0) \
```
Comment (Exairnous) [inline addons/io_hubs_addon/io/utils.py:371]:
```suggestion
    socket = lightmap_node.inputs.get("Lightmap") if bpy.app.version < (4, 1, 0) \
```

**Review by Exairnous — CHANGES_REQUESTED** [review]
Left a couple slight changes for 4.0 support, but otherwise it looks good to me.

**Review by Exairnous — APPROVED** [review]
LGTM.  Merging.

**Pull request merged**